lib/matplotlib/__init__.py

```diff
@@ -225,7 +225,7 @@ def _is_writable_dir(p):
             t.write(b'1')
         finally:
             t.close()
-    except OSError:
+    except:
```

Comment:
Ah, I didn't there was a separate `BaseException` under `Exception`, but all the more reason not to explicitly catch just `Exception`.  

I don't see how could could "raise strings"--that would result in a TypeError.